apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -231,6 +231,10 @@ type BMCDetails struct {
 	// insecure because it allows a man-in-the-middle to intercept the
 	// connection.
 	DisableCertificateVerification bool `json:"disableCertificateVerification,omitempty"`
+
+	// privLevel is the ipmi privilage level to used when authenicating, it is ignore
+	// when not using ipmi
+	PrivLevel string `json:"privLevel,omitempty"`
```

Comment:
I think we've so far been able to avoid driver-specific API fields, which helped avoid passing unused values to the driver factory and the internal API changes that triggers. 

Can this be encoded into the BMC URL somehow?

Reply:
Ya

> I think we've so far been able to avoid driver-specific API fields, which helped avoid passing unused values to the driver factory and the internal API changes that triggers.
> 
> Can this be encoded into the BMC URL somehow?

Yes, I believe it could, it was one of my earlier ideas but I instead implemented it in a similar way to "disableCertificateVerification" is there was a precedent. If encoding in the BMC url is preferred I'll push something up.

Reply:
`disableCertificateVerifcation` could potentially be used by any BMC driver (assuming it's not already), but IIUC this field only applies to IPMI, so they're not quite the same.

I would prefer using a URL parameter, but we should get input from @zaneb and others in the community (probably posting to the metal3-io version of this repo instead of the openshift version would help with that).

Reply:
Ack, thanks, will close in favour of discussing on 
https://github.com/metal3-io/baremetal-operator/pull/853